### PR TITLE
add HexTweaks, I think?

### PR DESCRIPTION
**Pull request by walksanatora**
If I understood correctly, this should finish implementing HexTweaks for HexBug

### CODE REVIEW

File: utils/mods.py [hunk 1]
```diff
@@ -9,13 +9,15 @@
...
-
 from Hexal.doc.collate_data import parse_book as hexal_parse_book
 from Hexal.doc.collate_data import pattern_stubs as hexal_stubs
 from HexMod.doc.collate_data import parse_book as hex_parse_book
 from HexMod.doc.collate_data import pattern_stubs as hex_stubs
 from MoreIotas.doc.collate_data import parse_book as moreiotas_parse_book
 from MoreIotas.doc.collate_data import pattern_stubs as moreiotas_stubs
+
+from HexTweaks.doc.collate_data import parse_book as hextweaks_parse_book
+from HexTweaks.doc.collate_data import pattern_stubs as hextweaks_stubs
```
Comment (object-Object) [inline utils/mods.py:20]: missing Black formatting

File: utils/mods.py [hunk 2]
```diff
@@ -240,6 +254,25 @@ class RegistryMod(Enum):
         pattern_stubs=moreiotas_stubs,
     )
 
+    HexTweaks = HexTweaksRegistryModInfo(
```
Comment (object-Object) [inline utils/mods.py:257]: missing `operator_directories`

File: cogs/info.py [hunk 3]
```diff
@@ -46,6 +46,10 @@ def from_mod(cls, mod: Mod) -> Self:
                 )
                 modloaders = [FORGE, FABRIC, QUILT]
 
+            case RegistryMod.HexTweaks:
+                description = "Adds various (mildly opinionated) quality of life changes, and dictionary iotas "
```
Comment (object-Object) [inline cogs/info.py:50]:
```suggestion
                description = "Adds various (mildly opinionated) quality of life changes, as well as dictionary iotas."
```

File: utils/mods.py [hunk 4]
```diff
@@ -240,6 +254,25 @@ class RegistryMod(Enum):
         pattern_stubs=moreiotas_stubs,
     )
 
+    HexTweaks = HexTweaksRegistryModInfo(
+        name="HexTweaks",
+        directory="HexTweaks",
+        book=hextweaks_parse_book(
+            "HexTweaks/common/src/main/resources",
+            "common/src/main/java",
+            "HexMod/Common/src/resources",
```
Comment (object-Object) [inline utils/mods.py:263]: Fixes to make the paths not invalid:
```suggestion
            "HexTweaks/common/src/main/java",
            "HexMod/Common/src/main/resources",
```

**Review by object-Object — CHANGES_REQUESTED** [review]
See comments. After fixing these issues, the bot still doesn't run because the HexTweaks repo has no version tags, so the version-fetching code fails.

File: utils/mods.py (hunk 2 above)
Comment (walksanatora) [reply utils/mods.py:257]: fixed

File: utils/mods.py (hunk 1 above)
Comment (walksanatora) [reply utils/mods.py]: resolved

File: utils/mods.py [hunk 5]
```diff
@@ -112,6 +114,18 @@ def _get_version(self) -> str:
         return next(filter(lambda t: self.version_regex.fullmatch(t), tags))
 
 
+@dataclass(kw_only=True)
+class HexTweaksRegistryModInfo(_BaseRegistryModInfo):
+    registry_regex: re.Pattern[str] = re.compile(
+        r'PatternRegistry.mapPattern\([\n ]+HexPattern\.fromAngles\("([qweasd]+)", ?HexDir\.(.+)?\)[,\n ]+?new ResourceLocation\(".+"(.+)?"\),\n.+,(.+)'
+    )
+    version_regex: re.Pattern[str] = re.compile(r"^\d+.\d+.\d+$")
```
Comment (object-Object) [inline utils/mods.py:122]:
```suggestion
    version_regex: re.Pattern[str] = re.compile(r"^v\d+.\d+.\d+$")
```

File: utils/mods.py [hunk 6]
```diff
@@ -112,6 +114,18 @@ def _get_version(self) -> str:
         return next(filter(lambda t: self.version_regex.fullmatch(t), tags))
 
 
+@dataclass(kw_only=True)
+class HexTweaksRegistryModInfo(_BaseRegistryModInfo):
+    registry_regex: re.Pattern[str] = re.compile(
+        r'PatternRegistry.mapPattern\([\n ]+HexPattern\.fromAngles\("([qweasd]+)", ?HexDir\.(.+)?\)[,\n ]+?new ResourceLocation\(".+"(.+)?"\),\n.+,(.+)'
```
Comment (object-Object) [inline utils/mods.py:120]: Registry regex is missing the classname. Also, the `is_great` bool group should only match if it's true — `"false"` coerces to True, which would make it treat every pattern as a great spell. See Hexal's regex for an example of both of these.

**Review by object-Object — CHANGES_REQUESTED** [review]

File: utils/mods.py (hunk 6 above)
Comment (walksanatora) [reply utils/mods.py:120]: need class name also? okay lemme quickly get that

Comment (walksanatora) [reply utils/mods.py]: added classname is now capture 4

File: utils/mods.py [hunk 7]
```diff
@@ -240,6 +254,26 @@ class RegistryMod(Enum):
...
+            "HexTweaks/common/src/main/resources",
+            "HexTweaks/common/src/main/java",
+            "HexMod/Common/src/main/resources",
+            "hextweaks",
+            "thetweakedbook",
+        ),
+        book_url="https://walksanatora.github.io/HexTweaks",
+        modrinth_url="https://modrinth.com/mod/hextweaks",
+        curseforge_url=None,
+        source_url="https://github.com/walksanatora/HexTweaks",
```
Comment (object-Object) [inline utils/mods.py:270]:
```suggestion
        book_url="https://walksanatora.github.io/HexTweaks/",
        modrinth_url="https://modrinth.com/mod/hextweaks/",
        curseforge_url=None,
        source_url="https://github.com/walksanatora/HexTweaks/",
```

**Comment by object-Object** [comment]
Also please run the book type scraper as detailed in the README, since your mod adds some extra fields

**Review by object-Object — APPROVED** [review]
thanks for the pr!

**Pull request merged**